src/libaktualizr/primary/aktualizr.cc

```diff
@@ -83,8 +84,24 @@ bool Aktualizr::UptaneCycle() {
 std::future<void> Aktualizr::RunForever() {
   std::future<void> future = std::async(std::launch::async, [&]() {
     SendDeviceData().get();
-    while (UptaneCycle()) {
-      std::this_thread::sleep_for(std::chrono::seconds(config_.uptane.polling_sec));
+
+    bool exiting = false;
+    std::mutex exit_m;
+    std::condition_variable exit_cv;
+    SigHandler::get().start([&exit_m, &exit_cv, &exiting]() {
```

Comment:
IMHO, a single handler should 'live' at an application level not at a library one since an application might incorporate not just libaktualizr but also another libraries and functionality that require triggering specific teardown actions on a system signal. Specifically, I would put a signal handler registration&instantion at aktualizr app (aktualizr-primary) level and triggers teardown functionality via Aktualizr API, it will allow an application developer to extend or override the signal handling functionality.